src/App.js

```diff
@@ -46,9 +47,8 @@ class App extends Component {
             actualWidth: 1
         };
         this.schematic = new PangenomeSchematic({store: this.props.store}); //Read file, parse nothing
-        observe(this.props.store, "beginBin", this.updateSchematicMetadata.bind(this));
+        observe(this.props.store, "beginEndBin", this.updateSchematicMetadata.bind(this));
         //TODO: endBin observer is still double triggering on shift() buttons. wait for beginBin event?
-        observe(this.props.store, "endBin", this.updateSchematicMetadata.bind(this));
         observe(this.props.store, "pixelsPerRow", this.recalcY.bind(this));
```

Comment:
The key problem I run into is that the observe function of `mobx` is not capable of deriving a complex MST model like observe an array. The `beginEndBin` in `ViewPortInputStore` are set correctly, but on change, the method `updateSchematicMetadata` is not triggered. 
There is a method `inject` in MST, but I haven no clue if that helps us and how to implement it.
I think that we will have to go away from the `observe` method and fine another way to trigger the method on a change.

Reply:
I found a solution for that. Just directly observe the object in the store and do not give a property.